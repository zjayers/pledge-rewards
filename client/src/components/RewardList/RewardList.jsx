import React from 'react';
import SubHeading from '../Shared/SubHeading/SubHeading';
import RewardList from './RewarsList.style';

export default (props) => {
  const { id, activated, rewardItems } = props;

  if (!activated) {
    return (
      <>
        <SubHeading uppercase>Includes:</SubHeading>
        <RewardList>
          {rewardItems.split(',').map((item, i) => (
            // eslint-disable-next-line react/no-array-index-key
            <li key={`${id}_${item}-${i}`}>{item}</li>
          ))}
        </RewardList>
      </>
    );
  }

  /* Return a Fragment */
  return <></>;
};
