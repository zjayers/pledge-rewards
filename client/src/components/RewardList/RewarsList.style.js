import styled from 'styled-components';

export default styled.ul`
  list-style: disc;
  padding-left: 1.2rem;
  margin-left: 1.8rem;
  margin-bottom: 20px;
`;
