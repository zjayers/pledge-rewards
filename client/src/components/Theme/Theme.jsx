import React from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
  colors: {
    borderGrey: '#9B9E9E',
    highlightGreen: '#009E74',
    buttonGreen: '#009E74',
    gradientBlue: '#DBE7FE',
    gradientOrange: '#FFF2EC',
    text: '#282828',
    flavorText: '#888B8B',
    linkText: '#498F80',
  },
  fonts: ['Muli', 'sans-serif'],
  fontSizes: {
    small: '1rem',
    medium: '1.8rem',
    large: '3rem',
  },
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;
