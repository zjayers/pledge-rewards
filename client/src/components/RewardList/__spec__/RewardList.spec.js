import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RewardList from '../RewardList';
import Theme from '../../Theme/Theme';

describe('RewardList Component', () => {
  let component;

  beforeEach(() => {
    component = mount(
      <Theme>
        <RewardList rewardItems='Test1,Test2,Test3' />
      </Theme>
    );
  });

  afterEach(() => {
    component.unmount();
  });

  it('should render without failure', () => {
    expect(component).toBeDefined();
  });

  it('should return am empty fragment if the "activated" prop is true', () => {
    component.unmount();
    component = shallow(
      <RewardList activated rewardItems='Test1,Test2,Test3' />
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should match the test snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
