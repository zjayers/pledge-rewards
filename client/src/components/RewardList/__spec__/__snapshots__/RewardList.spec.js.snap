// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RewardList Component should match the test snapshot 1`] = `
<Theme>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "borderGrey": "#9B9E9E",
          "buttonGreen": "#009E74",
          "flavorText": "#888B8B",
          "gradientBlue": "#DBE7FE",
          "gradientOrange": "#FFF2EC",
          "highlightGreen": "#009E74",
          "linkText": "#498F80",
          "text": "#282828",
        },
        "fontSizes": Object {
          "large": "3rem",
          "medium": "1.8rem",
          "small": "1rem",
        },
        "fonts": Array [
          "Muli",
          "sans-serif",
        ],
      }
    }
  >
    <_default
      rewardItems="Test1,Test2,Test3"
    >
      <_default
        uppercase={true}
      >
        <SubHeadingstyle___default
          uppercase={true}
        >
          <p
            className="SubHeadingstyle___default-qgikex-0 cDJTT"
          >
            Includes:
          </p>
        </SubHeadingstyle___default>
      </_default>
      <RewarsListstyle___default>
        <ul
          className="RewarsListstyle___default-gbe8yx-0 gRmdJF"
        >
          <li
            key="undefined_Test1-0"
          >
            Test1
          </li>
          <li
            key="undefined_Test2-1"
          >
            Test2
          </li>
          <li
            key="undefined_Test3-2"
          >
            Test3
          </li>
        </ul>
      </RewarsListstyle___default>
    </_default>
  </ThemeProvider>
</Theme>
`;

exports[`RewardList Component should return am empty fragment if the "activated" prop is true 1`] = `<Fragment />`;
